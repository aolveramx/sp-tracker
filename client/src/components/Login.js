import { LoginButton } from './styles/Login/LoginButton.styled'
import { HomeContainer } from './styles/Login/HomeContainer.styled'

const auth_url =
  'https://accounts.spotify.com/authorize?client_id=1fae6c9d97bd4d35b7771fd0a0ba8ba6&response_type=code&redirect_uri=http://localhost:3000&scope=streaming%20user-read-email%20user-read-private%20user-library-read%20user-library-modify%20user-read-playback-state%20user-modify-playback-state'

const Login = () => {
  return (
    <HomeContainer>
      <LoginButton href={auth_url}>Login with Spotify</LoginButton>
    </HomeContainer>
  )
}

export default Login
