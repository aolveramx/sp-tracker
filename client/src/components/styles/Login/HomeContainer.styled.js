import styled from 'styled-components'

export const HomeContainer = styled.div`
  align-items: center;
  display: flex;
  justify-content: center;
  max-width: 600px;
  margin: 40px auto;
  min-height: 50vh;
  overflow: hidden;
  padding: 20px;
`