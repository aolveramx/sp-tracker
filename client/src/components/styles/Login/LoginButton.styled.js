import styled from 'styled-components'

export const LoginButton = styled.a`
  background: #33CC33;
  border: 0;
  border-radius: 8px;
  color: #fff;
  cursor: pointer;
  padding: 8px;
  text-decoration: none;
  transition: all 0.5s;

  &:hover {
    color: #33CC33;
    background-color: #fff;
  }
`