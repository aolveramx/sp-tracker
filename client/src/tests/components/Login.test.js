import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import Login from '../../components/Login';


describe('Tests in Login.js', () => {

  test('should render <Login />', () => {

    const wrapper = shallow(<Login />)
    expect(wrapper).toMatchSnapshot()
    
  });
  
});