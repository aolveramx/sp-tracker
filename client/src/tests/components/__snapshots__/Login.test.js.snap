// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in Login.js should render <Login /> 1`] = `
<styled.div>
  <styled.a
    href="https://accounts.spotify.com/authorize?client_id=1fae6c9d97bd4d35b7771fd0a0ba8ba6&response_type=code&redirect_uri=http://localhost:3000&scope=streaming%20user-read-email%20user-read-private%20user-library-read%20user-library-modify%20user-read-playback-state%20user-modify-playback-state"
  >
    Login with Spotify
  </styled.a>
</styled.div>
`;
